constantine/math/elliptic/ec_twistededwards_affine.nim

```diff
@@ -76,6 +81,51 @@ func isOnCurve*[F](x, y: F): SecretBool =
   t2 -= t0
   return t2.isOne()
 
+func trySetFromCoordX*[F](P: var ECP_TwEdwards_Aff[F], x: F): SecretBool =
+  ## Try to create a point on the elliptic curve from X co-ordinate
+  ##   ax²+y²=1+dx²y²    (affine coordinate)
+  ## 
+  ## return true and update `P` if `y` leads to a valid point
+  ## return false otherwise, in that case `P` is undefined.
+  
+  # y² = (1 - ax²)/(1 - dx²)
+  var t {.noInit.}: F
+  var one {.noInit.}: F
+  one.setOne()
+
+  # (1 - dx²)
+  t.square(x)
+  when F.C.getCoefD() is int:
+    when F.C.getCoefD() >= 0:
+      P.y.fromUint uint F.C.getCoefD()
+    else:
+      P.y.fromUint uint -F.C.getCoefD()
+      P.y.neg()
+  else:
+    P.y = F.C.getCoefD()
+  P.y *= t
+  P.y.neg()
+  P.y += one
+
+  # (1 - ax²)
```

Comment:
a future TODO, a = 5 and we should shortcut when a <= 15 to use the addition chains here: https://github.com/mratsim/constantine/blob/7b64f85a29c466b9f3b827059b9587a7a39d979b/constantine/math/arithmetic/finite_fields.nim#L475-L479